File: src/com/silverkeytech/android_rivers/outlines/Head.java
package com.silverkeytech.android_rivers.outlines;

import com.silverkeytech.android_rivers.DateHelper;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

import java.util.Date;


public class Head {
    @Element
    public String title;
    @Element
    public String dateCreated;       //RFC822 time
    @Element
    public String dateModified;      //RFC822 time
    @Element
    public String ownerName;

    public Date getDateCreated() {
        if (dateCreated == null)
            return null;

        try {
            Date date = DateHelper.parseRFC822(dateCreated);
            return date;

        } catch (Exception e) {
            return null;
        }
    }

    public Date getDateModified() {
        if (dateModified == null)
            return null;

        try {
            Date date = DateHelper.parseRFC822(dateModified);
            return date;
        } catch (Exception e) {
            return null;
        }
    }
}

File: src/com/silverkeytech/android_rivers/outlines/Head.kt
package com.silverkeytech.android_rivers.outlines

import org.simpleframework.xml.Element
import hirondelle.date4j.DateTime
import com.silverkeytech.android_rivers.DateHelper

public class Head() {
    [Element] public var title: String? = null
    [Element] public var dateCreated: String? = null       //RFC822 time
    [Element] public var dateModified: String? = null      //RFC822 time
    [Element] public var ownerName: String? = null

    public fun getDateCreated() : java.util.Date?{
        if (dateCreated == null)
            return null

        var date = DateHelper.parseRFC822(dateCreated)

        return date
    }

    public fun getDateModified() : java.util.Date?{
        if (dateModified == null)
            return null

        var date = DateHelper.parseRFC822(dateModified)
        return date
    }
}
